src/features/user/TreeMapper/Analytics/components/Map/index.tsx

```diff
@@ -299,10 +300,19 @@ export const MapContainer = () => {
           return;
         }
       }
-      fetchProjectLocations();
     }
   }, [project, species, queryType, fromDate, toDate]);
 
+  useDebouncedEffect(
+    () => {
+      if (project && species) {
+        fetchProjectLocations();
+      }
+    },
+    500,
+    [project, species, queryType, fromDate, toDate]
```

Comment:
You maybe should change this paramter to `deps: any[]` in  `useDebouncedEffect`.

Reply:
It then won't trigger a call to `fetchProjectLocations` function anytime we change the filter value

Reply:
I did so now and do not see any problem. Can you explain this once more?

Reply:
Yes you're right, it won't cause any effect.